### DISCUSSION TITLE
Database size is not reduced after deleting records.

### DISCUSSION THREAD

**Discussion by rajagopalx** [simolus3/drift#1499] [Q&A] [answered]
I saw vacuum can be used to remove empty space after deletion.
Can you please help me how to execute the vacuum command in the moor/drift

Reference:
https://stackoverflow.com/questions/2143800/change-sqlite-file-size-after-delete-from-table

**Comment by simolus3** [comment, marked as answer]
You can just use `customStatement('VACUUM')` in your database to execute this command. 

If you want to enable the `auto_vacuum` pragma, a good place to do that is in the `beforeOpen` callback of your migration:

```dart
MigrationStrategy(
  // onCreate, onUpdate, ...
  beforeOpen: (details) async {
    await customStatement('pragma auto_vacuum = 1');
  }
)
```

**Reply by rajagopalx** [reply]
Thank You.
